src/media/js/elements/categories.js

```diff
@@ -0,0 +1,110 @@
+        prototype: Object.create(HTMLUListElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    var helperDiv = document.createElement('div');
+                    var catTemplate =
+                        '<mkt-category-item ' +
+                        ' data-mkt-category-name="{name}"' +
+                        ' data-mkt-category-slug="{slug}">' +
+                        '</mkt-category-item>';
+
+                    categories.forEach(function(cat) {
+                        helperDiv.innerHTML = format.format(catTemplate, {
+                            name: cat.name,
+                            slug: cat.slug
+                        });
+                        root.appendChild(helperDiv.firstChild);
+                    });
+                }
+            },
+            updateActiveNode: {
+                value: function(path) {
+                    var root = this;
+
+                    // Remove highlights from formerly-active nodes.
+                    var links = root.querySelectorAll('a.' + LINK_ACTIVE);
+                    for (var i = 0; links && (i < links.length); i++) {
+                        links[i].classList.remove(LINK_ACTIVE);
+                    }
+
+                    // Highlight new active nodes based on current page.
+                    var activeLinks = root.querySelectorAll(
+                        'a[href="' + (path || window.location.pathname) + '"]');
+                    for (i = 0; activeLinks && (i < activeLinks.length); i++) {
+                        activeLinks[i].classList.add(LINK_ACTIVE);
+                    }
+                }
+            }
+        })
+    });
+
+    var MktCategoryItemElement = document.registerElement('mkt-category-item', {
+        prototype: Object.create(HTMLLIElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+                    var name = root.getAttribute('data-mkt-category-name');
+                    var slug = root.getAttribute('data-mkt-category-slug');
+
+                    // Set slug on the item.
+                    root.setAttribute('data-mkt-category-item', slug);
+
+                    // Create <a> with href and title.
+                    var catLink = document.createElement('a');
+                    catLink.classList.add(LINK);
+                    catLink.setAttribute(
+                        'href', urls.reverse('category', [slug]));
+                    catLink.setAttribute('title', name);
+                    catLink.setAttribute('data-nav-no-active-node', '');
+                    catLink.textContent = name;
+                    root.appendChild(catLink);
+                }
+            },
+        })
+    });
+
+    z.page.on('loaded navigate', function() {
+        var catList = document.querySelector('mkt-category-list');
+        if (catList) {
```

Comment:
Can you check if `catList.updateActievNode` is `undefined` instead of wrapping this in a `try`/`catch`?